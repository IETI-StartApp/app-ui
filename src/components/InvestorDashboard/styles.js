import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(() => ({
    chart:{
        paddingRight: '20px',
    }
}));