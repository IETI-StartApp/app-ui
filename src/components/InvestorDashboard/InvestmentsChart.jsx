import React from 'react';
import {
    Chart,
    ArgumentAxis,
    ValueAxis,
    LineSeries,
} from '@devexpress/dx-react-chart-material-ui';
import { Paper } from '@material-ui/core';
import { useStyles } from './styles';
import { Animation } from '@devexpress/dx-react-chart';

export const InvestmentsChart = () => {
    const classes = useStyles();
    const data = [
        {
            Month: "Diciembre", Total: 0,
        }, {
            Month: "Enero", Total: 0,
        }, {
            Month: "Febrero", Total: 0,
        }, {
            Month: "Marzo", Total: 700000,
        }, {
            Month: "Abril", Total: 1800000,
        }, {
            Month: "Mayo", Total: 1000000,
        },
    ];

    return (
        <>
            <Paper>
                <Chart
                    data={data}
                    className={classes.chart}
                >
                    <ArgumentAxis />
                    <ValueAxis />
                    <LineSeries valueField="Total" argumentField="Month" color="#950740"/>
                    <Animation />
                </Chart>
            </Paper>
        </>
    );
}