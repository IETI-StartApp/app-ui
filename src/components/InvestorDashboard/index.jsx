import React from 'react';
import AttachMoneyIcon from '@material-ui/icons/AttachMoney';
import StarBorderIcon from '@material-ui/icons/StarBorder';
import TrendingUpIcon from '@material-ui/icons/TrendingUp';
import BookmarkBorderIcon from '@material-ui/icons/BookmarkBorder';
import { Dashboard } from '../Dashboard/index';
import { InvestmentsChart } from './InvestmentsChart';

export const InvestorDashboard = () => {
    const items = [
        { number: "4", description: "Proyectos Financiados", icon: <StarBorderIcon fontSize="large" />, path: "/" },
        { number: "3.5M", description: "Capital invertido", icon: <AttachMoneyIcon fontSize="large" />, path: "/dashboard" },
        { number: "1000", description: "Oportunidades de inversión ", icon: <TrendingUpIcon fontSize="large" />, path: "/" },
        { number: "2", description: "Proyectos guardados", icon: <BookmarkBorderIcon fontSize="large" />, path: "/" }
    ];
    const projects = [
        {
            name: "Proyecto 4", 
            image:"https://fundacioncompartir.org/sites/default/files/la-nueva-tecnologia-que-esta-moviendo-al-mundo.jpg", 
            finance: { value:15500000, valuation:30000000, startDate: "11/03/2021"}
        },
        {
            name: "Proyecto 5", 
            image:"https://i1.wp.com/tiempodenegocios.com/wp-content/uploads/2019/05/5798-750x406.jpg", 
            finance: { value:8000000, valuation:18000000, startDate: "15/02/2021"},
        },
        {
            name: "Proyecto 6", 
            image:"https://assets.entrepreneur.com/content/3x2/2000/tecnologia16.jpg", 
            finance: { value:8550000, valuation:12500000, startDate: "25/01/2021"},
        },
    ];


    return (
        <Dashboard items={ items } projects={ projects } chart={ <InvestmentsChart /> } projectsTitle={ "Oportunidades de inversión" } chartTitle={ " Inversiones en los últimos 6 meses" } />
    );
}