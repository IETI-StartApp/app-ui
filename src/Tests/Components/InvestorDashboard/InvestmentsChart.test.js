import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import { InvestmentsChart } from '../../../components/InvestorDashboard/InvestmentsChart';

describe('Testing Investments Chart', () => {

    const wrapper = shallow(<InvestmentsChart />)

    test('should load Investments Chart ', () => {
        expect(wrapper).toMatchSnapshot();
    })
})
