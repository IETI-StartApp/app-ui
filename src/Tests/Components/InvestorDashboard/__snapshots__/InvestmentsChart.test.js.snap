// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Investments Chart should load Investments Chart  1`] = `
<Fragment>
  <WithStyles(ForwardRef(Paper))>
    <Component
      className="makeStyles-chart-1"
      data={
        Array [
          Object {
            "Month": "Diciembre",
            "Total": 0,
          },
          Object {
            "Month": "Enero",
            "Total": 0,
          },
          Object {
            "Month": "Febrero",
            "Total": 0,
          },
          Object {
            "Month": "Marzo",
            "Total": 700000,
          },
          Object {
            "Month": "Abril",
            "Total": 1800000,
          },
          Object {
            "Month": "Mayo",
            "Total": 1000000,
          },
        ]
      }
    >
      <Component />
      <Component />
      <Component
        argumentField="Month"
        color="#950740"
        valueField="Total"
      />
      <AnimationBase
        duration={1000}
        easing={[Function]}
      />
    </Component>
  </WithStyles(ForwardRef(Paper))>
</Fragment>
`;
